tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaProduceSyncIntegration.cs

```diff
@@ -32,15 +32,24 @@ public class KafkaProduceSyncIntegration
         /// <typeparam name="TTarget">Type of the target</typeparam>
         /// <typeparam name="TTopicPartition">Type of the TopicPartition</typeparam>
         /// <typeparam name="TMessage">Type of the message</typeparam>
-        /// <typeparam name="TDeliveryHandler">Type of the delivery handler action</typeparam>
         /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
         /// <param name="topicPartition">TopicPartition instance</param>
         /// <param name="message">Message instance</param>
         /// <param name="deliveryHandler">Delivery Handler instance</param>
         /// <returns>Calltarget state value</returns>
-        internal static CallTargetState OnMethodBegin<TTarget, TTopicPartition, TMessage, TDeliveryHandler>(TTarget instance, TTopicPartition topicPartition, TMessage message, TDeliveryHandler deliveryHandler)
+        internal static CallTargetState OnMethodBegin<TTarget, TTopicPartition, TMessage>(TTarget instance, TTopicPartition topicPartition, TMessage message, ref object deliveryHandler)
```

Comment:
You shouldn't switch this to `object`, it could cause needless boxing - also, I don't know if it's a good idea to be using `ref` to change this, do we have to 🤔 or even if that does what you think it does 🤔

Reply:
I need to set a non-null delivery handler for this method call to get the produce offsets in case the use didn't set the delivery handler. This parameter is passed directly to the method, so I don't see any other way of modifying it.... If I stop using `object` here i will need to create a proper delegate here, which has generic args, which seems to be problematic. Anything you could recommend doing instead?

Reply:
> If I stop using object here i will need to create a proper delegate here

As mentioned above, you'll need to create a "proper" delegate either way unfortunately, so I don't think this gets you anything. In fact, using the `TDeliveryHandler` may be the only way you _can_ create a delegate of the correct type...